crates/stages/src/stages/hashing_account.rs

```diff
@@ -137,43 +175,57 @@ impl<DB: Database> Stage<DB> for AccountHashingStage {
         // AccountHashing table. Also, if we start from genesis, we need to hash from scratch, as
         // genesis accounts are not in changeset.
         if to_transition - from_transition > self.clean_threshold || stage_progress == 0 {
-            // clear table, load all accounts and hash it
-            tx.clear::<tables::HashedAccount>()?;
-            tx.commit()?;
-
-            let mut first_key = None;
-            loop {
-                let next_key = {
-                    let mut accounts = tx.cursor_read::<tables::PlainAccountState>()?;
-
-                    let hashed_batch = accounts
-                        .walk(first_key)?
-                        .take(self.commit_threshold as usize)
-                        .map(|res| res.map(|(address, account)| (keccak256(address), account)))
-                        .collect::<Result<BTreeMap<_, _>, _>>()?;
-
-                    let mut hashed_account_cursor = tx.cursor_write::<tables::HashedAccount>()?;
-
-                    // iterate and put presorted hashed accounts
-                    if first_key.is_none() {
-                        hashed_batch
-                            .into_iter()
-                            .try_for_each(|(k, v)| hashed_account_cursor.append(k, v))?;
-                    } else {
-                        hashed_batch
-                            .into_iter()
-                            .try_for_each(|(k, v)| hashed_account_cursor.insert(k, v))?;
-                    }
+            let mut checkpoint = self.get_checkpoint(tx)?;
+
+            if checkpoint.address.is_none() ||
+                checkpoint.to != to_transition ||
+                checkpoint.from != from_transition
+            {
+                // clear table, load all accounts and hash it
+                tx.clear::<tables::HashedAccount>()?;
+
+                // Checkpoint is invalid if the range changed
```

Comment:
what does this comment mean? seems unrelated?

Reply:
tried adding a better explanation and move it to a more relevant line. does it make more sense  ?